### PR TITLE
[azservicebus] Adding in the DeleteMessages feature. This is a new feature in preview.

### LINKED ISSUE #22800: [azservicebus] Implement the new DeleteMessages operation
DeleteMessages is a new feature, being added in preview, that lets you delete messages without having to transfer and Complete() them on the client side. 

- [ ] Build the code
- [ ] Add links back to documentation that show things like limits.

### PR DESCRIPTION
**Pull request by richardpark-msft**
Fixes #22800

### CODE REVIEW

File: sdk/messaging/azservicebus/receiver.go [hunk 1]
```diff
@@ -358,6 +364,83 @@ func (r *Receiver) DeadLetterMessage(ctx context.Context, message *ReceivedMessa
 	return r.settler.DeadLetterMessage(ctx, message, options)
 }
 
+// DeleteMessagesOptions contains the optional parameters for the [Client.DeleteMessages] method.
+type DeleteMessagesOptions struct {
+	// Count is the maximum number of messages to delete.
+	// Defaults to 4000.
+	Count int
```
Comment (jhendrixMSFT) [inline sdk/messaging/azservicebus/receiver.go:371]: Is there a max value for this that needs to be doc'ed?

File: sdk/messaging/azservicebus/receiver.go [hunk 2]
```diff
@@ -358,6 +364,83 @@ func (r *Receiver) DeadLetterMessage(ctx context.Context, message *ReceivedMessa
...
+	if options.Count == 0 {
+		options.Count = 4000
+	}
+
+	count := int64(0)
+
+	err := r.amqpLinks.Retry(ctx, EventReceiver, "DeleteMessages", func(ctx context.Context, lwid *internal.LinksWithID, args *utils.RetryFnArgs) error {
+		value := map[string]any{
+			"enqueued-time-utc": options.BeforeEnqueueTime.UTC(),
+			"message-count":     int32(options.Count),
```
Comment (jhendrixMSFT) [inline sdk/messaging/azservicebus/receiver.go:398]: This makes me wonder if `Count` should be an `int32`. Will this always be limited to a 32-bit value?

Comment (richardpark-msft) [reply sdk/messaging/azservicebus/receiver.go:398]: Let me check on that. I'll also adjust the Count to be an int64/int32 depending on this as well.

Comment (richardpark-msft) [reply sdk/messaging/azservicebus/receiver.go:398]: So it is limited to an int32 (same value but larger type fails). I've changed Count's type to reflect that.

File: sdk/messaging/azservicebus/receiver.go (hunk 1 above)
Comment (richardpark-msft) [reply sdk/messaging/azservicebus/receiver.go]: Currently it's 4000. However, do we normally document service-side limits on the client? It feels similar to client-side validation (ie, we should let the server control the messaging in it's error).

Comment (jhendrixMSFT) [reply sdk/messaging/azservicebus/receiver.go]: Good point. Maybe a link to the docs for this?

File: sdk/messaging/azservicebus/receiver.go (hunk 2 above)
Comment (jhendrixMSFT) [reply sdk/messaging/azservicebus/receiver.go:398]: The cast is now redundant.

**Review by jhendrixMSFT — APPROVED** [review]

File: sdk/messaging/azservicebus/receiver.go (hunk 1 above)
Comment (richardpark-msft) [reply sdk/messaging/azservicebus/receiver.go]: I added it as a task to the overall DeleteMessages issue (which I had no filed)
https://github.com/Azure/azure-sdk-for-go/issues/22800

**Pull request merged**